lib/routes/hitsz/due.ts

```diff
@@ -0,0 +1,158 @@
+
+export const handler = async (ctx) => {
+    const finalLimit = Number.parseInt(ctx.req.query('limit') ?? '20', 10);
+    const baseUrl = 'http://due.hitsz.edu.cn';
+
+    // 按类型分组（可直接用于 handler 中的逻辑判断）
+    const categoryGroups = {
+        // 教务核心业务（传统教务管理相关）
+        teaching: [
+            'jwxw/jwgl', // 教务管理
+            'jwxw/kwgl', // 考务管理
+            'jwxw/zcgl', // 注册管理
+            'jwxw/xkgl', // 选课管理
+            'jwxw/cjgl', // 成绩管理
+        ],
+
+        // 学籍相关（学生档案与身份管理）
+        studentStatus: [
+            'jwxw/xjgl_b_', // 学籍管理（本）
+            'jwxw/xjgl_y_', // 学籍管理（研）
+        ],
+
+        // 教学支持（辅助教学的资源与服务）
+        teachingSupport: [
+            'jwxw/jxxxh', // 教学信息化
+            'jwxw/jzxj', // 奖助学金
+        ],
+
+        // 学生培养（不同学段的培养动态）
+        education: [
+            'xwgl/bksxw', // 本科生新闻
+            'xwgl/ssxwpy/ktyzj', // 硕士学位培养
+            'xwgl/bsxwpy/qqhj1', // 博士学位培养
+        ],
+    };
+
+    // 修改：将 query 参数改为 path 参数（PR评论要求：Use path parameter instead of search query）
+    const { type = 'all' } = ctx.req.param(); // 从路径参数获取，默认值 'all'
+    const validTypes = Object.keys(categoryGroups);
+    const validType = validTypes.includes(type) ? type : 'all'; // 容错：无效类型默认走 all
+
+    // 根据类型选择对应栏目组
+    const categories = validType === 'all' ? Object.values(categoryGroups).flat() : categoryGroups[validType];
+
+    // 并发抓取所有栏目的第一页
+    const pagePromises = categories.map((category) => {
+        const pageUrl = new URL(`${category}.htm`, baseUrl).href;
+        return got(pageUrl).catch(() => null);
+    });
+
+    const pageResponses = await Promise.all(pagePromises);
+
+    // 提取所有文章链接
+    const articlePromises = [];
+    for (const [i, response] of pageResponses.entries()) {
+        if (!response) {
+            continue;
+        }
+
+        const category = categories[i];
+        const $ = load(response.data);
+        const listItemsOnPage = $('ul.box-main-list li, .list-main li, .list-main-modular li').toArray();
+
+        // 修改：将 for 循环 push 改为 map 生成数组（PR评论要求：Use map instead of push）
+        const categoryItems = listItemsOnPage
+            .map((el) => {
+                const $el = $(el);
+                const linkUrl = $el.find('a').attr('href');
+                if (!linkUrl) {return null;} // 过滤无链接项
+
+                const title = $el.find('span.text-over, a').text().trim();
+                const pubDateStr = $el.find('label').text().trim();
+
+                return {
+                    title,
+                    link: new URL(linkUrl, baseUrl).href,
+                    pubDate: pubDateStr ? timezone(parseDate(pubDateStr), 8) : null,
+                    category,
+                    description: title, // 使用标题作为描述
+                };
+            })
+            .filter(Boolean); // 过滤 null 项
+
+        articlePromises.push(...categoryItems); // 展开数组添加到承诺列表
+    }
+
+    // 获取所有文章详情
+    const allResolvedItems = (await Promise.all(articlePromises)).filter(Boolean);
```

Comment:
https://github.com/DIYgod/RSSHub/blob/b998b5ad9e79035a2eacd5c35716eabe69e79067/lib/routes/hitsz/due.ts#L59-L90 are synchronous which does not require Promise.all. Please use [flatMap](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap) instead.